src/app/services/project.service.ts

```diff
@@ -88,10 +88,22 @@ export class ProjectService {
     return this.api.getProject(projId, cpStart, cpEnd)
       .map((projects: Project[]) => {
         // get upcoming comment period if there is one and convert it into a comment period object.
+        // If there are multiple comment periods any that is currently running is a higher priority than a past comment period
         if (projects) {
-
-          if (projects[0].commentPeriodForBanner && projects[0].commentPeriodForBanner.length > 0) {
+          if (projects[0].commentPeriodForBanner && projects[0].commentPeriodForBanner.length === 1) {
             projects[0].commentPeriodForBanner = new CommentPeriod(projects[0].commentPeriodForBanner[0]);
+          } else if (projects[0].commentPeriodForBanner && projects[0].commentPeriodForBanner.length > 1) {
+              let now = new Date
+              let currentDate = now.toISOString();
+              // Default to the same comment period we're using currently in case one is not active
+              let finalCommentPeriod = new CommentPeriod(projects[0].commentPeriodForBanner[0]);
+              for ( let commentPeriod in projects[0].commentPeriodForBanner){
+                if (Date.parse(projects[0].commentPeriodForBanner[commentPeriod].dateCompleted) > Date.parse(currentDate)
```

Comment:
Date.parse(projects[0].commentPeriodForBanner[commentPeriod].dateCompleted

So... if projects, projects[0] are null, or Date.parse fails, this will crash. The .pipe below has a catch, but might be worth putting in a check for null/bad values. Just a if(project && project[0]) should do the trick. Otherwise it's working as expected.